homeassistant/components/rfxtrx/__init__.py

```diff
@@ -235,16 +243,29 @@ def handle_receive(event):
 
         _LOGGER.debug("Receive RFXCOM event: %s", event_data)
 
-        data_bits = get_device_data_bits(event.device, device_bits)
+        data_bits = get_device_data_bits(event.device, devices)
         device_id = get_device_id(event.device, data_bits=data_bits)
 
         # Callback to HA registered components.
         hass.helpers.dispatcher.dispatcher_send(SIGNAL_EVENT, event, device_id)
 
         # Signal event to any other listeners
-        if device_id in device_events:
+        fire_event = devices.get(device_id, {}).get(CONF_FIRE_EVENT)
+        if fire_event:
             hass.bus.fire(EVENT_RFXTRX_EVENT, event_data)
 
+    @callback
+    def device_update(event, device_id):
+        if device_id not in devices:
+            data = entry.data.copy()
+            event_code = binascii.hexlify(event.data).decode("ASCII")
+            data[CONF_DEVICES][event_code] = device_id
+            hass.config_entries.async_update_entry(entry=entry, data=data)
+            devices[device_id] = {}
```

Comment:
Where will the config dict be populated? In a future options flow?

Reply:
Yes, by default detected devices have no options set and the options flow will handle setting properties that are now being set in yaml

Reply:
Yes, by default detected devices have no options set and the options flow will handle setting properties that are now being set in yaml

Reply:
See #37982, it's work in progress, shouldn't take long to finish once this last bit of preparation is in place

Reply:
Ok. :+1: